fix missing utilities in Windows wheels

benoit-pierre opened the pull request:
Some utilities, like `rcc.exe`, were getting filtered-out: handlethe fact that those are "not a Qt executable", as reported by `windeployqt`, but we still want to include them.

Close #21.

altendky commented on build.py:
```diff
@@ -1185,20 +1188,28 @@ def windeployqt_list_source(
-        )
-    except subprocess.CalledProcessError as e:
-        raise DependencyCollectionError(target) from e
+    process = report_and_check_call(
+        command=[
+            windeployqt,
+            '--dry-run',
+            '--list', 'source',
+            target,
+        ],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        # ugh, 3.5
+        # encoding='utf-8',
+        check=False,
+    )
+
+    if process.returncode:
+        if (process.returncode > 0 and
+            'does not seem to be a Qt executable'
+            in process.stderr.decode('utf-8')):
+            # This can happen with some utilities like
+            # `rcc.exe`, which we still want to include.
+            return []
+        raise DependencyCollectionError(target, process)
```
It looks like we can just pull the return code and stderr off the exception.

```suggestion
    try:
        process = report_and_check_call(
            command=[
                windeployqt,
                '--dry-run',
                '--list', 'source',
                target,
            ],
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
            # ugh, 3.5
            # encoding='utf-8',
        )
    except subprocess.CalledProcessError as e:
        if 'does not seem to be a Qt executable' in e.stderr.decode('utf-8'):
            # This can happen with some utilities like
            # `rcc.exe`, which we still want to include.
            return []
 
        raise DependencyCollectionError(target) from e
```

benoit-pierre replied:
Yes, much better.

altendky commented on build.py:
```diff
@@ -714,10 +714,13 @@ def report_and_check_call(command, *args, cwd=None, shell=False, **kwargs):
         for arg in command:
             print('    {}'.format(repr(arg)))
 
+    if 'check' not in kwargs:
+        kwargs['check'] = True
+
```
With the other suggestion, this isn't needed anymore.  But, if it were, or is in the future, it could just be added as a parameter default above similar to how `cwd` is handled.

```suggestion
```

altendky commented on build.py:
```diff
@@ -714,10 +714,13 @@ def report_and_check_call(command, *args, cwd=None, shell=False, **kwargs):
     if cwd is not None:
         cwd = fspath(cwd)
-    return subprocess.run(command, *args, cwd=cwd, check=True, **kwargs)
+    return subprocess.run(command, *args, cwd=cwd, **kwargs)
```
As above.

```suggestion
    return subprocess.run(command, *args, cwd=cwd, check=True, **kwargs)
```

altendky reviewed (commented):
Thanks for taking the time to identify the solution here.  Hopefully we don't end up missing deps.  Other platforms don't have `windeployqt` so at some point I should just stop using it entirely I guess...

I think the important bits of your changes can be added within the existing structure.  Let me know what you think of my suggested form.

Cheers,
-kyle

benoit-pierre committed:
fix missing utilities in Windows wheels

Some utilities, like `rcc.exe`, were getting filtered-out: handle
the fact that those are "not a Qt executable", as reported by
`windeployqt`, but we still want to include them.

benoit-pierre commented:
The checks have passed, but the Github Actions run is still failed!?

And the error is less than helpful:
```
CI : .github#L1
Strategy produced more than 256
```

benoit-pierre commented:
Too many matrix combinations for the test job?

benoit-pierre commented:
> A job matrix can generate a maximum of 256 jobs per workflow run. This limit also applies to self-hosted runners.

Ref: https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix

altendky commented:
Yes, I have started to trim that down.  At least in https://github.com/altendky/qt-applications/pull/20, I forget exactly.

benoit-pierre commented:
Maybe you could start by dropping Python 3.5 support, since it reached EOL a year ago.

s-weigand commented:
Any news on this? 😄

altendky commented:
Merging as is and will explore the results in fresh builds in https://github.com/altendky/qt-applications/pull/23.  Thank you for the work and I apologize for the delayed resolution.
